integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/TestUtils.java

```diff
@@ -178,6 +178,28 @@ public static synchronized int getNextFreePort(int from, int to) {
     return port;
   }
 
+  private static int port = 30000;
+  private static final int END_PORT = 32767;
+
+  /**
+   * Get the next free port between port and END_PORT.
+   *
+   * @return the next free port number, if there is no free port below END_PORT return -1.
+   */
+  public static synchronized int getNextFreePort() {
+    LoggingFacade logger = getLogger();
+    int freePort = 0;
+    while (port <= END_PORT) {
+      freePort = port++;
+      if (isLocalPortFree(freePort)) {
```

Comment:
Even though this method is synchronized, I'm concerned that you will still have timing bugs. I could see this method returning a given port value, but there being a delay before the test uses that port and it no longer being free. That would allow the method to return the same port to multiple callers. What does the isLocalPortFree method do?

Reply:
I checked... It's what I thought. I believe that this is still not safe.

Reply:
Isn't the port value increased every time the method is called? So once allocated that port is never reused in the life of the test run.

Reply:
If the port was used by some other process before used by the tests yes it won't work.

Reply:
It doesn't seem to. The logic always starts at 0 and looks for the next port not currently in use.  Once it returns a certain port number there will be some delay before that port is in use since it's not recorded here. Even though you've had some successful test runs, this code isn't safe. There only seem to be about 50 ports. Why not just assign port values to each test?